src/Perpetuum/Zones/Terrains/Materials/Minerals/MineralNode.cs

```diff
@@ -50,6 +50,13 @@ public uint[] Values
             get { return _values; }
         }
 
+        public event MineralNodeEventHandler Decrease;
+
+        private void OnDecrease()
+        {
+            Decrease?.Invoke(this);
+        }
```

Comment:
Not sure why we want a private method that does this? Doesn't seem to serve a purpose over a direct call from the public method.

Reply:
Seems to be a pattern to extract logic that is not strictly part of the returning value or control flow. Seen it elsewhere, like NPC's have a lot of `On___()` methods that may do some internal state changes and maybe some event firings.
In this case, it is strictly event firings, but mostly I was following the pattern set by: `MineralNode.OnExpired()` `MineralNode.OnUpdated()`.

Reply:
Maybe it is verbose just for an event firing, but makes it consistent with the other events.